code/clue.js

```diff
@@ -1,59 +1,325 @@
+  poison,
+  bat,
+  throphy,
+  pistol,
+];
 
-const rooms = []
+const rooms = [
+  "Dinning Room",
+  "Conservatory",
+  "Conservatory",
+  "Study",
+  "Study",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio",
+];
+
+//Console log the arrays
+console.log(suspects);
+console.log(weapons);
+console.log(rooms);
+
+//Console log and play around with console logging the first suspect in the suspects array or the last room in the rooms array
+console.log(suspects[0]);
+console.log(suspects[1].firstName);
+console.log(weapons[5].name);
+console.log(rooms[rooms.length - 1]);
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
 // YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
-const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+const randomSelector = (array) => {
+  return array[Math.floor(Math.random() * array.length)];
+};
 
 // CREATE AN OBJECT THAT KEEPS THE MYSTERY.
 // With a killer, a weapon and a room.
 // The values will be set later.
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null,
+};
+
+// This function will randomly change favourite weapons of each person
+const shuffleFavouriteWeapon = () => {
+  suspects.forEach((suspect) => {
+    const randomWeapon = randomSelector(weapons);
+    suspect.favouriteWeapon = randomWeapon.id;
+  });
+};
+
+const cleanRevealMysteryMessage = () => {
+  document.getElementById("mystery").innerHTML = "";
+};
+
+const toggleAnimationDeckCard = (id) => {
+  document.getElementById(id).classList.toggle("deck-heartbeat");
+};
 
 // This function will be invoked when you click on the killer card.
-const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+const pickKiller = (event) => {
+  event.preventDefault();
+  if (!killerPicked) {
+    // This will remove the reveal the crime message if any
+    cleanRevealMysteryMessage();
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+    toggleAnimationDeckCard("killerCard");
+
+    setTimeout(() => {
+      shuffleFavouriteWeapon();
+
+      // This will randomly select a killer from the suspects. And add that to the mystery object.
+      mystery.killer = randomSelector(suspects);
+
+      // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
```

Comment:
I think you can delete a few blank lines here, I find it difficult to follow the code with the blank lines inside of a function